DOCSP-36546 Scan Multiple Collections (#193)
mongodb/docs-spark-connector: source/streaming-mode/streaming-read-config.txt

   * - ``collection``
     - | **Required.**
       | The collection name configuration.

>        | You can specify multiple collections by separating each collection name

Reviewer: ```suggestion
       | You can specify multiple collections by separating the collection names
```

>        | You can specify multiple collections by separating each collection name

Reviewer: S: Probably worth mentioning somewhere that you can only stream from multiple collections in a single database (if I'm reading the Jira ticket correctly)

>          with a comma (,). Do not add a space between the collections

Reviewer: ```suggestion
         with a comma. Do not add a space between the collections
```

         unless the space is a part of the collection name.
       | Specify multiple collections as follows:

       .. code-block:: java

          ...
          .option("spark.mongodb.collection", "collectionOne,collectionTwo")

>        | You can scan from all collections in the database by using an asterisk (*).

Reviewer: S: I think "stream" is probably a better and more accurate verb here

```suggestion
       | You can stream from all collections in the database by using an asterisk (*).
```

>        | You can scan from all collections in the database by using an asterisk (*).
>        | Specify all collections as follows:

Reviewer: ```suggestion
       | You can scan from all collections in the database by passing an asterisk (*) as a string for the collection name.
       | Specify all collections as follows:
```

feel free to play with this part; i think it's worth it to specify that it needs to be in quotes (though the compiler should catch this)

       .. code-block:: java

          ...
          .option("spark.mongodb.collection", "*")

>        If you create a collection while scanning from all collections, it is
>        automatically picked up for scanning.

Reviewer: ```suggestion
       If you create a collection while streaming from all collections, the new collection
       is automatically included in the stream.
```

>    * - ``collection``

Reviewer 2: The following information provided in "Documentation Changes Summary" is not reflected anywhere:

> When scanning from multiple/all collections, collections can be freely dropped while being scanned.

Author: I moved the content from this table into [its own section here](https://preview-mongodbjordansmith721.gatsbyjs.io/spark-connector/DOCSP-36546-multiple-collections-support/streaming-mode/streaming-read-config/#specifying-multiple-collections-in-the-collection-property) and added that information

Reviewer 2: This is resolved, but I don't have the technical ability to resolve the discussion.

>    * - ``collection``

Reviewer 2: The information from the "Escaping" section of "Documentation Changes Summary" is missing from the documentation of the `collection` configuration option. Those rules should be specified here in full.

P.S. This comment does not imply any changes to `release-notes.txt` in addition to what has already been done.

Author: Added to the configuration options page

Reviewer 2: Let's not forget to specify this part:

> Note that if the value is specified as a string literal in Java code, then each reverse solidus has to be further escaped, leading to having to specify "\\\\".

[optional]
I think examples (like those given in "Documentation Changes Summary") are helpful and recommend (but don't insist, nor can I) including them.

Author: Updated to add examples and add the note for Java string literals

Reviewer 2: ✅ This is resolved, but I don't have the technical ability to resolve the discussion.

     - | **Required.**
       | The collection name configuration.
       | You can specify multiple collections by separating the collection names

>          with a comma. The collections must be in the same database. Do not add

Reviewer 2: ```suggestion
         with a comma. Do not add
```

This sentence brings only confusion: there is no way to configure the connector to read from different databases (see the `database` configuration option), and the `collection` configuration option allows users to specify only the collection names, not namespaces (they include the database name). So this sentence rises questions like "How could I possibly specify collections from multiple databases?", "Does this mean there is a way to read from multiple databased via the connector?".

Author: Removed

  .. note:: 
     
     When specifying the collection name as a string literal in Java, you must
     further escape each backslash with another one. For example, escape a collection 
     named "\\collection" as follows:

     .. code-block:: java

        "\\\\collection"

You can stream from all collections in the database by passing an
asterisk (*) as a string for the collection name.

Specify all collections as shown in the following example:

.. code-block:: java

   ...
   .option("spark.mongodb.collection", "*")

If you create a collection while streaming from all collections, the new
collection is automatically included in the stream. 

You can drop collections at any time while streaming from multiple collections.

.. important:: Inferring the Schema of a Change Stream

   If you set the ``change.stream.publish.full.document.only``
   option to ``true``, the {+connector-short+} infers the schema of a ``DataFrame``
   by using the schema of the scanned documents. If you set the option to
   ``false``, you must specify a schema.
   
   Schema inference happens at the beginning of streaming, and does not take into
   account collections that are created during streaming.

>    When streaming from multiple collections, the connector samples
>    each collection sequentially. Streaming from a large number of

Reviewer 2: [optional]

My original wording in the "Documentation Changes Summary" was proven to be unclear, and now I am afraid to leave users confused. I am suggesting the following additions:

```suggestion
   When streaming from multiple collections, and inferring the schema,
   the connector samples each collection sequentially
   as part of the schema inference. Streaming from a large number of
```

Author: Added to this to make it extra clear

collection is automatically included in the stream. 

You can drop collections at any time while streaming from multiple collections.

.. important:: Inferring the Schema of a Change Stream

   If you set the ``change.stream.publish.full.document.only``
   option to ``true``, the {+connector-short+} infers the schema of a ``DataFrame``
   by using the schema of the scanned documents. If you set the option to

>    ``false``, you must specify a schema.

Reviewer 2: [optional]

This part of the note duplicates the note in `streaming-read.txt`. While I find it weird to have such duplication, it's up to the docs team. However, the source code for both notes is also duplicated, which opens opportunities to update one of the notes and not the other, thus causing confusion.

Author: There were a few minor differences in the two different admonitions, so unfortunately I couldn't single-source them. If this does begin causing issues we can look at rewording/moving things (or maybe just removing one of the duplicates)
